lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleRadioSelect/stateful/__tests__/NxStatefulCollapsibleRadioSelect.test.tsx

```diff
@@ -4,96 +4,521 @@
+    });
+
+    it('clears the input when the Escape key is pressed', async function() {
+      const user = userEvent.setup(),
+          inputEl = quickFilterRender().getByRole('textbox');
+
+      inputEl.focus();
+      await user.keyboard('b');
+      expect(inputEl).toHaveValue('b');
+
+      await user.keyboard('[Escape]');
+      expect(inputEl).toHaveValue('');
+    });
+
+    it('clears the input when clear filter button is clicked', async function() {
+      const user = userEvent.setup(),
+          view = quickFilterRender(),
+          inputEl = view.getByRole('textbox');
+
+      inputEl.focus();
+      await user.keyboard('b');
+      await runTimers();
+      expect(inputEl).toHaveValue('b');
+
+      const clearBtn = view.getByRole('button', { name: 'Clear filter' });
+      await user.click(clearBtn);
+
+      expect(inputEl).toHaveValue('');
+    });
+
+    it('calls preventDefault when the ESC is pressed and filter value is not empty', async function() {
+      const user = userEvent.setup(),
+          inputEl = quickFilterRender().getByRole('textbox'),
+          keyEvent = createEvent.keyDown(inputEl, { key: 'Escape' });
+
+      inputEl.focus();
+      await user.keyboard('b');
+
+      fireEvent(inputEl, keyEvent);
+
+      expect(keyEvent.defaultPrevented).toBe(true);
+    });
+
+    it('doesn\'t call preventDefault when the ESC is pressed and filter value is empty', async function() {
+      const inputEl = quickFilterRender().getByRole('textbox'),
+          keyEvent = createEvent.keyDown(inputEl, { key: 'Escape' });
+
+      inputEl.focus();
+      fireEvent(inputEl, keyEvent);
+
+      expect(keyEvent.defaultPrevented).toBe(false);
+    });
+
+    describe('when options are filtered', function() {
+      it('renders all options if the text input value is an empty string', function() {
+        const view = quickFilterRender(),
+            inputEl = view.getByRole('textbox'),
+            options = view.getByRole('menu').children;
+
+        expect(inputEl).toHaveValue('');
+        expect(options.length).toBe(4);
+
+        expect(options[0]).toHaveTextContent('Foo');
+        expect(options[1]).toHaveTextContent('Bar');
+        expect(options[2]).toHaveTextContent('Baz');
+        expect(options[3]).toHaveTextContent('Boo');
+      });
+
+      it('renders filtered options when the text input value changes, ignoring case sensitivity', async function() {
+        const user = userEvent.setup(),
+            view = quickFilterRender(),
+            inputEl = view.getByRole('textbox'),
+            options = view.getByRole('menu').children;
+
+        expect(options.length).toBe(4);
+
+        inputEl.focus();
+        await user.keyboard('ba');
+
+        expect(options.length).toBe(2);
+        expect(options[0]).toHaveTextContent('Bar');
+        expect(options[1]).toHaveTextContent('Baz');
+
+        await user.keyboard('[Escape]');
+        expect(inputEl).toHaveValue('');
```

Comment:
Maybe assert that the filtered-out options came back at this point

Reply:
> Maybe assert that the filtered-out options came back at this point

added here: [913c519](https://github.com/sonatype/sonatype-react-shared-components/pull/932/commits/913c5198b613c2afdf30d7e903f72c28d2cf5841)